cmd/lotus-worker/main.go

```diff
@@ -130,12 +135,34 @@ var stopCmd = &cli.Command{
 
 		ctx := lcli.ReqContext(cctx)
 
-		// Detach any storage associated with this worker
+		// If forcefully flag is not set, perform a graceful shutdown
+		if !cctx.Bool("forcefully") {
+			// Step 1: Disable new tasks being scheduled
+			for taskType := range allowSetting {
+				if err := api.TaskDisable(ctx, taskType); err != nil {
+					return err
+				}
+			}
+
+			// Step 2: Wait for existing jobs to complete
+			// Repeat 30 times to ensure all tasks have finished
+			for i := 0; i < 30; i++ {
+				time.Sleep(time.Second)   // wait for 1 second
+				err := api.WaitQuiet(ctx) // wait for tasks to finish
+				if err != nil {
+					log.Errorf("Error waiting for tasks to finish: %v", err)
+					return err
+				}
+			}
```

Comment:
If graceful shutdown fails repeatedly I would just fail the stop command and print that user should run it with --force flag, or alternatively print that graceful shutdown failed and that we will do a forced shutdown

Reply:
Yeah, the tricky with trying to get to a graceful shutdown is that:

- Sealing work that is currently running on a lotus-worker can take hours before its done. (the current loop might be to spammy/aggresive)
   - If the worker just finished a PC1-tasks, but is also schedule to do that sectors PC2-task, we want be sure not to shut it down when its "apperently" not busy doing anything in that transition. 

I think users would want this graceful shutdown command as a way to schedule a shutdown of some worker for maintenance, when its finished with its current work.